### PR TITLE
Parse job configurations for easier usage analytics

### PR DESCRIPTION
**Pull request by wagnermarkd**
This Pull Requests adds functionality to parse job configurations into an Avro Map. This is added as an additional job at the beginning of ProccessLogs. As part of this work, a "CombineDocumentFileFormat" is introduced which returns key values as <filename, document bytes>.

### CODE REVIEW

File: hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                
+        job.setOutputKeyClass(BytesWritable.class);
+        job.setOutputValueClass(NullWritable.class);
+  
+        job.setInputFormatClass(CombineDocumentFileFormat.class);
+        job.setOutputFormatClass(AvroKeyOutputFormat.class);
+  
+        AvroJob.setOutputKeySchema(job, JobConf.SCHEMA$);
+        
+        job.setNumReduceTasks(0);
```
Comment (matthayes) [inline hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java:148]: why setting to 0?

File: hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import org.xml.sax.SAXException;
+
+import com.linkedin.whiteelephant.mapreduce.lib.input.CombineDocumentFileFormat;
+import com.linkedin.whiteelephant.mapreduce.lib.job.StagedOutputJob;
+import com.linkedin.whiteelephant.mapreduce.lib.job.StagedOutputJobExecutor;
+import com.linkedin.whiteelephant.util.JobStatsProcessing;
+
+public class ParseJobConfs
+{  
+  private static final String CPU_MILLISECONDS = "CPU_MILLISECONDS";
```
Comment (matthayes) [inline hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java]: not used

**Comment by matthayes** [comment]
Looks pretty good overall, thanks!  just wondering about the num reducers equaling 0.  What does this do?

File: hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java (hunk 1 above)
Comment (wagnermarkd) [inline hadoop/src/java/com/linkedin/whiteelephant/parsing/ParseJobConfs.java:148]: This is a map only job so I need to set it to 0 to indicate that

**Comment by wagnermarkd** [comment]
I've removed the unused field. Let me know if there's anything else.

**Pull request merged**